src/course7/homework/Challenge7.java

```diff
@@ -0,0 +1,27 @@
+package course7.homework;
+
+import java.util.Scanner;
+
+public class Challenge7 {
+
+    public static void main(String[] args) {
+
+        Scanner scan = new Scanner(System.in);
+
+        System.out.println("Guess a number between 0 and 10.");
+
+        double guess;
+
+        do {
+            System.out.println("Enter a number:");
+            guess = scan.nextDouble();
+
+            if (guess < 5) {
+                System.out.println("Too low.");
+            } else if (guess > 5) {
+                System.out.println("Too high.");
+            }
+        } while (guess != 5);
```

Comment:
for clarity and easier to read (best practice) you could define a constant of this number 5 and reuse it, in order to understand better the intentions:
```
public class Challenge7 {
  static final int GUESSED_NUMBER = 5; // this makes the variable a constant

  public static void main(String[] args) {
    ...
    do {
      ... // use GUESSED_NUMBER everywhere instead of number 5
    } while (guess != GUESSED_NUMBER);
    ...
  }
}
```

Reply:
Fair point. To be honest, I didn't use a separate variable for the number 5 because it seemed redundant at that point, since all I had to do with the value got from the use was to compare it with 5 or any other value of my choosing within a certain . I've now made some alterations to "force" the user to stay within the specified range :)).